registry/quilt_server/core.py

```diff
@@ -86,12 +88,14 @@ def __json__(self):
 class FileNode(Node):
     json_type = 'FILE'
 
-    def __init__(self, hashes, metadata=None):
-        if metadata is None:
-            metadata = {}
-
+    def __init__(self, hashes, metadata):
         assert isinstance(hashes, list)
         assert isinstance(metadata, dict)
+        assert metadata.get('q_path')
+        assert not os.path.isabs(metadata['q_path'])
+
+        if not metadata.get('q_path'):
```

Comment:
Won't happen cause you already have an assert above

Reply:
*nod* assert added afterward, and I didn't remove that.  Done.

Reply:
I'll post these changes tomorrow -- it's late here.